enhancements/local-storage/kubesan-csi-driver-integration-into-lvms.md

```diff
@@ -0,0 +1,452 @@
+  - Perform stress tests to evaluate system behavior under high load and failure conditions.
+  - We will run these tests before any graduation to GA at the minimum.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- **Developer Preview (Early Evaluation and Feedback)**:
+  - Initial implementation with basic functionality for shared and node-local VG provisioning.
+  - Basic integration and E2E tests in place.
+  - Feedback from early adopters and stakeholders collected.
+  - No official Product Support.
+  - Functionality is provided with very limited, if any, documentation. Documentation is not included as part of the product’s documentation set.
+
+- **Technology Preview**:
+  - Feature-complete implementation with all planned functionality.
+  - Comprehensive test coverage including performance and stress tests.
+  - Functionality is documented as part of the products documentation set (on the Red Hat Customer Portal) and/or via the release notes.
+  - Functionality is provided with LIMITED support by Red Hat. Customers can open support cases, file bugs, and request feature enhancements. However, support is provided with NO commercial SLA and no commitment to implement any changes.
+  - Functionality has undergone more complete Red Hat testing for the configurations supported by the underlying product.
+  - Functionality is, with rare exceptions, on Red Hat’s product roadmap for a future release.
+
+### Tech Preview -> GA
+
+- **GA**:
+  - Proven stability and performance in production-like environments.
+  - Positive feedback from initial users.
+  - Full documentation, including troubleshooting guides and best practices.
+  - Full LVMS Support Lifecycle
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+- **Upgrade**:
+  - Ensure that upgrades are seamless with no downtime for existing workloads. Migrating to a kubesan enabled version is a no-break operation
+  - Test upgrade paths thoroughly to ensure compatibility and data integrity. The kubesan to topolvm (or vice versa) switch should be excluded and forbidden explicitly.
+  - The "default" deviceClass cannot be changed as well and changeing from shared to local or vice versa is not supported without resetting the LVMCluster.
+  - New deviceClasses with the shared policy should be able to be added to existing LVMClusters without affecting existing deviceClasses.
+
+- **Downgrade**:
+  - Allow safe downgrades by maintaining backward compatibility. Downgrading from a kubesan enabled version to a purely topolvm enabled version should be a no-break operation for the topolvm part. For the kubesan part, the operator should ensure that the shared VGs can be cleaned up manually
+  - Provide rollback mechanisms and detailed instructions to revert to previous versions. Ensure that downgrades do not result in data loss or service interruptions.
```

Comment:
It would be nice if we had a more detailed description of how data loss during rollbacks will be avoided (this is just stating that we will do so). I don't think we necessarily need to have a complete design in this EP, but I think that having a complete story for rollback and DR should be part of the graduation criteria for GA.

Reply:
I added a rollback routine along with both automated as well as manual expected rollback procedures and impacts on data in the design details section.